database/scrapers/building_scrapper.ts

```diff
@@ -0,0 +1,158 @@
+interface CampusesData {
+  data: {
+    name: string;
+    cover: string;
+    buildings: string[];
+  }[];
+}
+
+export default class BuildingScrapper extends BaseScraperModule {
+  name = "building and campuses scrapper";
+  description =
+    "scrapes pwr buildings data from directus and campuses from local file: './assets/campuses.json'";
+
+  private filesService = new FilesService();
+
+  async run(task: TaskHandle) {
+    try {
+      this.logger.info("starting reading campuses file...");
+      const campusesData = JSON.parse(
+        await fs.readFile("./assets/campuses.json", { encoding: "utf-8" }),
+      ) as CampusesData | undefined;
+      if (campusesData === undefined) {
+        throw new Error("failed reading ./assets/campuses.json");
+      }
+      const campusesMap = new Map<string, Campus>();
+      for (const data of campusesData.data) {
+        const campus = await Campus.create(data);
+        data.buildings.forEach((building) => campusesMap.set(building, campus));
+      }
+      this.logger.info("campuses created!");
+
+      this.logger.info("starting fetching buildings...");
+      const response = await fetch(buildingsPath);
+      if (!response.ok) {
+        throw new Error(
+          `Error: ${response.status} ` + `cause: ${response.statusText}`,
+        );
+      }
+      const buildingsBody = (await response.json()) as
+        | { data: BuildingsData[] }
+        | undefined;
+      if (buildingsBody === undefined) {
+        throw new Error("failed fetching buildings: empty body");
+      }
+      const buildingsData = buildingsBody.data;
+
+      const formattedBuildingData = await Promise.all(
+        buildingsData.map(async (data) => {
+          const addressArray = data.addres.split(",");
+          return {
+            id: data.id,
+            identifier: data.name,
+            specialName: data.naturalName,
+            iconType: BuildingIcon.Icon,
+            campusId: -1,
+            addressLine1: addressArray.shift(),
+            addressLine2: addressArray.shift(),
+            latitude: data.latitude,
+            longitude: data.longitude,
+            haveFood: data.food ?? false,
+            cover: await this.uploadAndGetKey(data),
+            externalDigitalGuideMode: data.externalDigitalGuideMode,
+            externalDigitalGuideIdOrURL: data.externalDigitalGuideIdOrURL,
+            createdAt: resolveDate(data.createdAt),
+            updatedAt: resolveDate(data.updatedAt),
+          };
+        }),
+      );
+      for (const buildingEntry of formattedBuildingData) {
+        const campus = campusesMap.get(buildingEntry.identifier);
+        if (campus === undefined) {
+          this.logger.error(
+            `No campus assigned to building ${buildingEntry.identifier}`,
+          );
+        } else {
+          // update campuses
+          campus.cover = buildingEntry.cover;
+          buildingEntry.campusId = campus.id;
+          await campus.save();
+        }
+      }
```

Comment:
just a minor code style nitpick: consider using `continue` in the if statement, so that you can avoid using an else branch.
... or maybe you should throw instead? If you continue without assigning the campusId, the database operating will just fail due to the foreign key constraint not being met.